### PR TITLE
ttyplot: 1.5.2 -> 1.6.1

### PR DESCRIPTION
**Pull request by Lassulus**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/tools/misc/ttyplot/default.nix [hunk 1]
```diff
@@ -1,21 +1,20 @@
...
   pname = "ttyplot";
-  version = "1.5.2";
+  version = "1.6.1";
 
   src = fetchFromGitHub {
     owner = "tenox7";
     repo = "ttyplot";
     rev = version;
-    sha256 = "sha256-BYMdGNDl8HUin1Hu4Fqgx305a/tTt1fztqlT2vDeTh8=";
+    sha256 = "sha256-SQ5keCcwzQsSxfSevQwRa1eNf+8JXsrh1vljehI4tPc=";
```
Comment (kirillrdy) [inline pkgs/tools/misc/ttyplot/default.nix:11]:
```suggestion
    hash = "sha256-SQ5keCcwzQsSxfSevQwRa1eNf+8JXsrh1vljehI4tPc=";
```

Comment (Lassulus) [reply pkgs/tools/misc/ttyplot/default.nix]: lightning fast review :D

Comment (kirillrdy) [reply pkgs/tools/misc/ttyplot/default.nix]: haha, just lucky timing

**Review by kirillrdy — APPROVED** [review]
Result of `nixpkgs-review pr 280602` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>ttyplot</li>
  </ul>
</details>

builds and runs

File: pkgs/tools/misc/ttyplot/default.nix [hunk 2]
```diff
@@ -1,21 +1,20 @@
...
 
-  buildInputs = [ ncurses ];
-
-  buildPhase = ''
-   ${stdenv.cc}/bin/cc ./ttyplot.c -lncurses -o ttyplot
-  '';
+  buildInputs = [
+    pkg-config
+    ncurses
+  ];
```
Comment (hartwork) [inline pkgs/tools/misc/ttyplot/default.nix:17]: @Lassulus now that the build phases uses `make` — great! —, the install phase probably should do too, in particular to also install the man page (which has a ton of non-trivial examples). There three relevant lines from the `Makefile` are:

```
MANPREFIX ?= $(PREFIX)/man
```
…and…
```
        install -d $(DESTDIR)$(MANPREFIX)/man1
        install -m644 ttyplot.1 $(DESTDIR)$(MANPREFIX)/man1
```

Comment (hartwork) [reply pkgs/tools/misc/ttyplot/default.nix:17]: @Lassulus PS: `licenses.unlicense;` does not look like Apache 2. Is that intentional?

Comment (Lassulus) [reply pkgs/tools/misc/ttyplot/default.nix:17]: ah, I guess back in the day there was no license yet when I packaged it

**Pull request merged**